File: solution/binary_search/10815/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/9e4cc6a176c6444d99cbc0379c5fd53a
#include<bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N; cin >> N;
    vector<int> V(N);
    for(auto &i: V) cin >> i;
    sort(V.begin(), V.end());
    int Q; cin >> Q;
    for(int i=0;i<Q;i++) {
        int x;cin >> x;
        if(binary_search(V.begin(), V.end(), x)) cout << 1 << ' ';
        else cout << 0 << ' ';
    }

    return 0;
}

File: solution/binary_search/10815/main.java
// Authored by : lms0806
// Co-authored by : -
// Link : http://boj.kr/7105d879677a4ebb96ade9837f789f03
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    static int[] arr;
    public static void main(String[] args) throws IOException{
        FastReader rd = new FastReader();

        int size = rd.nextInt();

        arr = new int[size];

        for(int i = 0; i < size; i++) {
            arr[i] = rd.nextInt();
        }

        Arrays.sort(arr);

        size = rd.nextInt();

        StringBuilder sb = new StringBuilder();
        while(size --> 0) {
            sb.append(binary_search(0, arr.length - 1, rd.nextInt())).append(" ");
        }
        System.out.print(sb);
    }

    public static int binary_search(int start, int end, int n) {
        while(start <= end) {
            int mid = (start + end) / 2;
            if(arr[mid] == n) {
                return 1;
            }

            if(n < arr[mid]) end = mid - 1;
            else start = mid + 1;
        }
        return 0;
    }

    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while(st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                }
                catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() { return Integer.parseInt(next()); }
        long nextLong() { return Long.parseLong(next()); }
        double nextDouble() { return Double.parseDouble(next()); }
        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            }
            catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}
